### DISCUSSION TITLE
AWS S3 Configuration via Roles

### DISCUSSION THREAD

**Discussion by pxslip** [directus/directus#7781] [Feature Requests]
Referencing the same issue discussed in https://github.com/directus/v8-archive/issues/710 I was curious whether similar functionality might be implemented in v9? Specifically, is it possible to either set an env variable, or by the lack of certain env variables have the AWS CLI use the role-based permissions? I'll add that I am happy to take a stab at this should it be a desirable feature.

**Comment by rijkvanzanten** [comment]
Role based permissions? 

FWIW currently, we pass every env var that starts with `STORAGE_<LOCATION>` on as a camelcased object to the AWS S3 library constructor, so anything that's available in there is theoretically available from Directus:

https://github.com/directus/directus/blob/4f284e455ec2597dae498aa5ca11092bdf2d563a/api/src/storage.ts#L30

https://github.com/directus/directus/blob/4f284e455ec2597dae498aa5ca11092bdf2d563a/packages/drive-s3/src/AmazonWebServicesS3Storage.ts#L44-L48

That being said, I don't fully know what you're talking about and therefore haven't tried it myself 🙂

**Reply by pxslip** [reply]
Yep, am a big fan of that approach to the config. Quick description of Role-Based permissions on an EC2: An EC2 can be granted permissions as if it were an AWS user, and any actions against the AWS APIs get signed using automatically generated credentials (makes rotating of key/secret unnecessary).

All of that being said, I am occasionally stupid and hadn't done enough debugging. The issue was in our bucket configuration, not Directus. So for future users who find this discussion - role-based permissions work perfectly, just don't set the `STORAGE_<LOCATION>_KEY`/`STORAGE_<LOCATION>_SECRET` and you should be good to go.

Thanks for your response!

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._

**Comment by TreeKat71** [comment]
Although key&secret works fine, I believe implementing this feature is crucial. Creating a user for the application to use could pose operational challenges and raise security concerns.
